tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/document/documents.controller.js

```diff
@@ -1400,117 +1401,133 @@
             vm.gridOptions.data = [];
             $scope.gridApi.infiniteScroll.resetScroll();
             if (sortColumns.length != 0) {
-                if (sortColumns[0].name == vm.gridOptions.columnDefs[1].name) {
-                    if (sortColumns[0].sort != undefined) {
-                        if (vm.FileNameSort == undefined || vm.FileNameSort == "asc") {
-                            vm.FileNameSort = "desc";
-                            vm.documentSortBy(vm.configSearchContent.ManagedPropertyFileName, 0, sortColumns[0].name, sortColumns[0].field, "asc");
+                if (sortColumns[0].name != undefined) {
+                    if (sortColumns[0].name.trim().toLowerCase() == configs.search.searchColumnsUIPickerForDocument.documentName.keyName.trim().toLowerCase()) {

```

Comment:
Reading the values from config.js, instead of reading from the grid column headers